api-v2/src/student/dto/create-student.dto.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  IsEmail,
+  IsNotEmpty,
+  IsInt,
+  IsString,
+  Max,
+  Min,
+  IsOptional,
+  IsObject,
+} from 'class-validator';
+import { ScheduleCourse } from '../../../../common/types';
+
+export class CreateStudentDto {
+  @IsNotEmpty()
+  @IsString()
+  fullName: string;
+
+  @IsString()
+  nuid: string;
+
+  @IsNotEmpty()
+  @IsEmail()
+  email: string;
+
+  @IsNotEmpty()
+  @IsString()
+  password: string;
+
+  @IsOptional()
+  @IsInt()
+  @Min(1898) // the year NEU was established
+  @Max(3000) // will GraduateNU last beyond year 3000?!
+  academicYear?: number;
+
+  @IsOptional()
+  @IsInt()
+  @Min(1898)
+  @Max(3000)
+  graduateYear?: number;
+
+  @IsOptional()
+  @IsInt()
+  @Min(1898)
+  @Max(3000)
+  catalogYear?: number;
+
+  @IsOptional()
+  @IsString()
+  major?: string;
+
+  @IsOptional()
+  @IsString()
+  coopCycle?: string;
+
+  @IsOptional()
+  @IsObject()
+  coursesCompleted?: ScheduleCourse[];
+
+  @IsOptional()
+  @IsObject()
+  coursesTransfered?: ScheduleCourse[];
+
+  @IsOptional()
+  @IsInt()
+  primaryPlanId?: number;
```

Comment:
Do we assign a primary ID when a student is created?

Reply:
hmm we don't assign a primaryPlanId initially since the student doesn't have a plan when created. this again has to do with how our onboarding works so we can change when we figure out onboarding!